api/v2/routes_organizations.go

```diff
@@ -167,135 +143,144 @@ func (api *API) registerOrgUser(c *gin.Context) {
 	}
-	var url string
-	// format the url the user clicks to activate email
-	if dev {
-		url = fmt.Sprintf(
-			"https://dev.api.temporal.cloud/v2/account/email/verify/%s/%s",
-			user.UserName, token,
-		)
-	} else {
-		url = fmt.Sprintf(
-			"https://api.temporal.cloud/v2/account/email/verify/%s/%s",
-			user.UserName, token,
-		)
-
+	// validate that the user is part of the organization
+	// however dont fail on an error, simply continue
+	usr, err := api.um.FindByUserName(forms["user"])
+	if err != nil {
+		api.LogError(c, err, eh.UserSearchError)
+		return
 	}
-	// format a link tag
-	link := fmt.Sprintf("<a href=\"%s\">link</a>", url)
-	emailSubject := fmt.Sprintf(
-		"%s Temporal Email Verification", forms["organization_name"],
-	)
-	// build email message
-	es := queue.EmailSend{
-		Subject: emailSubject,
-		Content: fmt.Sprintf(
-			"please click this %s to activate temporal email functionality", link,
-		),
-		ContentType: "text/html",
-		UserNames:   []string{user.UserName},
-		Emails:      []string{user.EmailAddress},
+	if usr.Organization != forms["name"] {
+		Fail(c, errors.New("user is not part of organization"))
+		return
 	}
-	// send email message to queue for processing
-	if err = api.queues.email.PublishMessage(es); err != nil {
-		api.LogError(c, err, eh.QueuePublishError)(http.StatusBadRequest)
+	var uploads []models.Upload
+	paged, err := gpaginator.Paging(
+		&gpaginator.Param{
+			DB:    api.upm.DB.Where("user_name = ?", forms["user"]),
+			Page:  pageInt,
+			Limit: limitInt,
+		},
+		&uploads,
+	)
+	if err != nil {
+		api.LogError(c, err, "failed to get paged user upload")
 		return
 	}
-	// log
-	api.l.With(
-		"user", forms["username"],
-		"organization", forms["organization_name"],
-		"organization.owner", username,
-	).Info("organization user account registered")
-	// remove hashed password from output
-	user.HashedPassword = "scrubbed"
-	// remove the verification token from output
-	user.EmailVerificationToken = "scrubbed"
-	// format a custom response that includes the user model
-	// and an additional status field
-	var status string
-	if dev {
-		status = fmt.Sprintf(
-			"by continuing to use this service you agree to be bound by the following api terms and service %s",
-			devTermsAndServiceURL,
-		)
-	} else {
-		status = fmt.Sprintf(
-			"by continuing to use this service you agree to be bound by the following api terms and service %s",
-			prodTermsAndServiceURL,
-		)
+	// return the response
+	Respond(c, http.StatusOK, gin.H{"response": paged})
+}
+
+func (api *API) getUploads(orgName string, users []string) (map[string][]models.Upload, error) {
+	resp := make(map[string][]models.Upload)
+	for _, user := range users {
+		uplds, err := api.orgs.GetUserUploads(orgName, user)
+		if err != nil {
+			continue
+		}
+		resp[user] = uplds
+	}
+	return resp, nil
+}
+
+// returns true if user is owner
+func (api *API) validateOrgOwner(c *gin.Context, organization, username string) (*models.Organization, bool) {
```

Comment:
I think returning (*models.Organization, error) is better, where err == nil is ok. so the error can be handled by the caller.

Reply:
There is no need to handle the error here. The only "handle" is return early and give an error.